File: 2021/day08/part1.js
const fs = require("fs").promises;

async function main() {
  const input = await fs.readFile("input.txt", "utf8");
  const lines = input.split("\n").map((l) => l && l.split(" | ")[1]);

  let qty = 0;

  for (let line of lines) {
    let words = line.split(" ");

    for (let word of words) {
      if ([2, 3, 4, 7].includes(word.length)) qty++;
    }
  }

  console.log(qty);
}

main();

File: 2023/day01/part1.py
def read_file():
  with open('input.txt', encoding="utf-8") as f:
    return f.read()

def collab_values(line):
  start, end = "", ""
  numbers = "0123456789"

  for i in range(len(line)):
    if line[i] in numbers:
      if start == "":
        start = line[i]
      else:
        end = line[i]
                   
  return start, end or start

def main():
  input = read_file()
  lines = input.split("\n")

  total = 0
  for line in lines:
    if line:
      start, end = collab_values(line)
      print(start, end)
      total += int(start + end)

  print(total)
    
if __name__ == '__main__':
  main()